backend/Controllers/MetadataController.cs

```diff
@@ -209,5 +206,64 @@ public async Task<IActionResult> PostAsync([FromBody] SaveMetadataResource resou
             return Unauthorized("Invalid permissions!");
 		}
 
+		/// <summary>
+		/// Adds a new DataSource, and connects it to a given Metadata.
+		/// Ensures the user is a member of the correct municipality, therby ensuring only relevant people can add DataSources.
+		/// </summary>
+		/// <param name="newDataSource">The data for the new DataSource to add.</param>
+		/// <returns>HTTP response code</returns>
+		[HttpPut("url")]
+		public async Task<IActionResult> PutUrlAsync([FromBody] NewDataSourceResource newDataSource)
+		{
+			if (!ModelState.IsValid)
+				return BadRequest(ModelState.GetErrorMessages());
+
+			var username = httpContextRetriever.HttpContext.User.Identity.Name;
+			var user = await userService.GetUserByMailAsync(username);
+			var metadata = await _metadataService.GetByUuidAsync(newDataSource.MetadataUuid);
+
+            if(user.MunicipalityName != metadata.MunicipalityName)
+				return Unauthorized("Invalid permissions for given Metadata! User must match Municipality.");
+
+			DataSource dataSource = _mapper.Map<NewDataSourceResource, DataSource>(newDataSource);
+
+            if(dataSource.StartDate == null && dataSource.EndDate != null)
+            {
+				return BadRequest("Cannot have startDate as null, but a set endDate");
```

Comment:
Maybe a better wording on this error message?

Reply:
Hmmm, do you have any suggestions?